netmiko/cisco_base_connection.py

```diff
@@ -179,6 +178,16 @@ def telnet_login(
                     return_msg += output
 
                     # is at spitfire xr prompt
+
+                    if re.search('RP/\d+/RP\d+/CPU\d+:\S*\)#$', output):
+                        # To check if it is in the config prompt
+                        # If yes, then write newline char, "clear" and "end" to clear any stale configs
+                        self.write_channel(self.TELNET_RETURN + "clear" + self.TELNET_RETURN)
+                        self.read_channel()
+                        self.write_channel("end" + self.TELNET_RETURN)
+                        output = self.find_prompt()
+                        return_msg += output
```

Comment:
I think it is gonna be dangerous as we will always have this prompt and try to clear it.
If we cant verify this is 1st time login , then we dont want to add this code..

I would rather focus on yes/no being passed.